tests/testthat/test-events.R

```diff
@@ -0,0 +1,293 @@
+test_that("create_event_based_processes assigns a listener to each event", {
+
+  pop <- get_population("Antigua and Barbuda")
+  contact_matrix_set <- squire::get_mixing_matrix(iso3c = "ATG")
+  parameters <- get_parameters(pop, 1, 100, contact_matrix_set)
+
+  events <- create_events()
+  states <- create_states(parameters)
+  max_age <- 100
+  variables <- create_variables(pop, max_age)
+  indiv <- create_human(states, variables, events)
+
+  create_event_based_processes(
+    indiv$human,
+    states,
+    variables,
+    events,
+    parameters
+  )
+
+  for (event in events) {
+    expect_gt(length(event$listeners), 0)
+  }
+
+})
+
+test_that("test that listeners do not call empty targets", {
```

Comment:
This tests the opposite. We call the listeners with empty targets and queue_state_update is called